modules/gdnative/godot/godot_basis.cpp

```diff
@@ -28,188 +28,245 @@
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
 #include "godot_basis.h"
+#include "core/variant.h"
 
-#include "math/matrix3.h"
+#include "core/math/matrix3.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-void _basis_api_anchor() {
+void _basis_api_anchor() {}
+
+void GDAPI godot_basis_new_with_rows(godot_basis *r_dest, const godot_vector3 *p_x_axis, const godot_vector3 *p_y_axis, const godot_vector3 *p_z_axis) {
+	const Vector3 *x_axis = (const Vector3 *)p_x_axis;
+	const Vector3 *y_axis = (const Vector3 *)p_y_axis;
+	const Vector3 *z_axis = (const Vector3 *)p_z_axis;
+	Basis *dest = (Basis *)r_dest;
+	*dest = Basis(*x_axis, *y_axis, *z_axis);
 }
 
-void GDAPI godot_basis_new(godot_basis *p_v) {
-	Basis *v = (Basis *)p_v;
-	*v = Basis();
+void GDAPI godot_basis_new_with_axis_and_angle(godot_basis *r_dest, const godot_vector3 *p_axis, const godot_real p_phi) {
+	const Vector3 *axis = (const Vector3 *)p_axis;
+	Basis *dest = (Basis *)r_dest;
+	*dest = Basis(*axis, p_phi);
 }
 
-void GDAPI godot_basis_new_with_euler_quat(godot_basis *p_v, const godot_quat *p_euler) {
-	Basis *v = (Basis *)p_v;
-	Quat *euler = (Quat *)p_euler;
-	*v = Basis(*euler);
+void GDAPI godot_basis_new_with_euler(godot_basis *r_dest, const godot_vector3 *p_euler) {
+	const Vector3 *euler = (const Vector3 *)p_euler;
+	Basis *dest = (Basis *)r_dest;
+	*dest = Basis(*euler);
 }
 
-void GDAPI godot_basis_new_with_euler(godot_basis *p_v, const godot_vector3 p_euler) {
-	Basis *v = (Basis *)p_v;
-	Vector3 *euler = (Vector3 *)&p_euler;
-	*v = Basis(*euler);
+godot_string GDAPI godot_basis_as_string(const godot_basis *p_self) {
+	godot_string ret;
+	const Basis *self = (const Basis *)p_self;
+	memnew_placement(&ret, String(*self));
+	return ret;
 }
 
-void GDAPI godot_basis_new_with_axis_and_angle(godot_basis *p_v, const godot_vector3 p_axis, const godot_real p_phi) {
-	Basis *v = (Basis *)p_v;
-	const Vector3 *axis = (Vector3 *)&p_axis;
-	*v = Basis(*axis, p_phi);
+godot_basis GDAPI godot_basis_inverse(const godot_basis *p_self) {
+	godot_basis dest;
+	const Basis *self = (const Basis *)p_self;
+	memnew_placement(&dest, Basis(self->inverse()));
```

Comment:
You only changed the ones I commented on, but I meant this for all the occurrences of `memnew_placement()` in that context. I just didn't want to write it 200 times 😄 

So yeah, please fix the rest too :D